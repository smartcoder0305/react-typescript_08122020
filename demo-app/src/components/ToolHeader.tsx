import React, { FC } from 'react';

import './ToolHeader.css';

export type ToolHeaderProps = {
  headerText?: string,
};

export const ToolHeader: FC<ToolHeaderProps> = (props) => {

  return (
    <header>
      <h1 className="page-header">{props.headerText}</h1>
    </header>
  );

};

ToolHeader.defaultProps = {
  headerText: 'The Tool',
};