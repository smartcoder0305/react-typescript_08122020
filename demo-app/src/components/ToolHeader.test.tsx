import * as React from 'react';
import {
  render, mount, shallow,
  ReactWrapper, ShallowWrapper,
} from 'enzyme';

import { ToolHeader } from './ToolHeader';

describe('<ToolHeader /> Enzyme Static HTML', () => {
  test('<ToolHeader /> renders', () => {
    const component = JSON.stringify(
      render(<ToolHeader headerText="The Tool" />).html(),
    );

    expect(component).toMatchSnapshot();
  });
});

describe('<ToolHeader /> Enzyme Mock DOM', () => {
  let component: ReactWrapper;
  let componentDOMNode: ReactWrapper;

  beforeEach(() => {
    component = mount(<ToolHeader headerText="The Tool" />);
    componentDOMNode = component.find('h1');
  });

  test('<ToolHeader /> renders', () => {
    expect(componentDOMNode.text()).toBe('The Tool');
  });
});

describe('<ToolHeader /> Shallow with Enzyme', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<ToolHeader headerText="The Tool" />);
  });

  test('<ToolHeader /> renders', () => {
    const h1Content = wrapper.find('h1').text();
    expect(h1Content).toBe('The Tool');
  });
});
