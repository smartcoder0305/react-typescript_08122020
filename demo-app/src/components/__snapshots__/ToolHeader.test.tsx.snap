// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolHeader /> Enzyme Static HTML <ToolHeader /> renders 1`] = `"\\"<h1 class=\\\\\\"page-header\\\\\\">The Tool</h1>\\""`;
